components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/services/ChallengeQuestionManagementAdminService.java

```diff
@@ -281,4 +278,35 @@ private void checkCrossTenantAccess(String tenantDomain) throws IdentityRecovery
 
     }
 
+    private boolean isValidUser(String userDomain, String tenantAwareUserName, String loggedInName) {
+
+        String loggedInUserDomain = UserCoreUtil.getDomainFromThreadLocal();
+        if (StringUtils.isEmpty(loggedInUserDomain)) {
+            loggedInUserDomain = UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME;
+        }
+        if (StringUtils.isEmpty(userDomain)) {
+            userDomain = UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME;
+        }
+        String domainAppendedUsername = appendDomainToUserName(tenantAwareUserName, userDomain);
+        String domainAppendedLoggedInName = appendDomainToUserName(loggedInName, loggedInUserDomain);
+        int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+        boolean isUsernameCaseSensitive = IdentityUtil.isUserStoreCaseSensitive(loggedInUserDomain, tenantId);
+
+        if (isUsernameCaseSensitive) {
+            return domainAppendedUsername.equals(domainAppendedLoggedInName);
+        } else {
+            return domainAppendedUsername.equalsIgnoreCase(domainAppendedLoggedInName);
+        }
+    }
+
+    private String appendDomainToUserName(String username, String domainName) {
```

Comment:
Don't we have an existing util to do this task ? Probably we should have one in identity.core

Reply:
This method has two cases. It will either convert the user store domain name into uppercase if it's already appended or append the name directly. For appending part I have called the UserCoreUtil method.

Reply:
https://github.com/wso2/carbon-identity-framework/blob/master/components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java#L759

Doesn't this do the same thing ?